opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -8,27 +10,79 @@
 _GCP_METADATA_URL_HEADER = {"Metadata-Flavor": "Google"}
 
 
-def get_gce_resources():
-    """ Resource finder for common GCE attributes
-
-        See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
-    """
+def _get_google_metadata_and_common_attributes():
     token = attach(set_value("suppress_instrumentation", True))
     all_metadata = requests.get(
         _GCP_METADATA_URL, headers=_GCP_METADATA_URL_HEADER
     ).json()
     detach(token)
-    gce_resources = {
-        "host.id": all_metadata["instance"]["id"],
+    common_attributes = {
         "cloud.account.id": all_metadata["project"]["projectId"],
-        "cloud.zone": all_metadata["instance"]["zone"].split("/")[-1],
         "cloud.provider": "gcp",
-        "gcp.resource_type": "gce_instance",
+        "cloud.zone": all_metadata["instance"]["zone"].split("/")[-1],
     }
-    return gce_resources
+    return common_attributes, all_metadata
+
+
+def get_gce_resources():
+    """ Resource finder for common GCE attributes
+
+        See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
+    """
+    (
+        common_attributes,
+        all_metadata,
+    ) = _get_google_metadata_and_common_attributes()
+    common_attributes.update(
+        {
+            "host.id": all_metadata["instance"]["id"],
+            "gcp.resource_type": "gce_instance",
+        }
+    )
+    return common_attributes
+
+
+def get_gke_resources():
+    """ Resource finder for GKE attributes
+
+    """
+    # The user must specify the container name via the Downward API
+    container_name = os.getenv("CONTAINER_NAME")
+    if container_name is None:
+        return {}
+    (
+        common_attributes,
+        all_metadata,
+    ) = _get_google_metadata_and_common_attributes()
+
+    # Fallback to reading namespace from a file is the env var is not set
+    pod_namespace = os.getenv("NAMESPACE")
+    if pod_namespace is None:
+        with open(
+            "/var/run/secrets/kubernetes.io/serviceaccount/namespace", "r"
+        ) as namespace_file:
+            pod_namespace = namespace_file.read().strip() or ""
```

Comment:
What is the `or ""` need for? If the file doesn't exist, you'd get an exception in the `open()` call, so if you want to handle that case you should do it explicitly